pkg/block/block.go

```diff
@@ -63,12 +63,22 @@ func Download(ctx context.Context, logger log.Logger, bucket objstore.Bucket, id
 	return nil
 }
 
-// Upload uploads block from given block dir that ends with block id.
+// Upload uploads a TSDB block to the object storage. It verifies basic
+// features of Thanos block and updates `meta.Thanos.File` section.
+func Upload(ctx context.Context, logger log.Logger, bkt objstore.Bucket, bdir string) error {
+	return upload(ctx, logger, bkt, bdir, true)
+}
+
+// UploadPromBlock uploads a TSDB block to the object storage. It assumes
+// the block is used in Prometheus so it doesn't check Thanos external labels.
+func UploadPromBlock(ctx context.Context, logger log.Logger, bkt objstore.Bucket, bdir string) error {
+	return upload(ctx, logger, bkt, bdir, false)
+}
+
+// upload uploads block from given block dir that ends with block id.
 // It makes sure cleanup is done on error to avoid partial block uploads.
-// It also verifies basic features of Thanos block.
 // TODO(bplotka): Ensure bucket operations have reasonable backoff retries.
-// NOTE: Upload updates `meta.Thanos.File` section.
-func Upload(ctx context.Context, logger log.Logger, bkt objstore.Bucket, bdir string) error {
+func upload(ctx context.Context, logger log.Logger, bkt objstore.Bucket, bdir string, thanosBlocks bool) error {
```

Comment:
I would propose being more explicit. Just "isThanosBlock` might be too vague and leaving room to surprises.

I would suggest either `checkExternalLabels bool`  (and upload debug meta for now in this cases too - it will be removed soon) OR maybe better: We can detect if it's Thanos block by checking meta.json if it has `Thanos section`, no? 🤔  (: 
```suggestion
func upload(ctx context.Context, logger log.Logger, bkt objstore.Bucket, bdir string, thanosBlocks bool) error {
```

Reply:
Agree. I updated the name. PTAL.